src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  getAlive() {
+    return this.health > 0;
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
[CODE PERFORMANCE] The `die` method uses `indexOf` and `splice` to remove the animal from `Animal.alive`. Consider using `filter` to avoid nesting loops and improve performance. For example: `Animal.alive = Animal.alive.filter(animal => animal.health > 0);` This could be done after taking damage in the `takeDamage` method or in a separate cleanup method.